_posts/2018-07-23-performance-profiling-with-craft-cms.md

```diff
@@ -0,0 +1,117 @@
+        <img src="{{ image.url }}" alt="{{ image.title }}">
+      {% endif %}
+    {% endfor %}
+  {% endraw %}
+{% endhighlight %}
+
+#### With Eager Loading
+{% highlight twig %}
+  {% raw %}
+    {% set 
+      entries = craft.entries()
+      .section('info') // 1. fetch entries from info section
+      .with(['imageField']) // 2. what should be Eager Loaded?
+      .all() 
+    %} 
+
+    {% for entry in entries %}
+      {% set image = entry.imageField[0] %} // 3. fetch the image
+      {% if image %}
+        <img src="{{ image.url }}" alt="{{ image.title }}">
+      {% endif %}
+    {% endfor %}
+  {% endraw %}
+{% endhighlight %}
+
+There are 2 major things you will notice:
+* While selecting all entries from a section (`info`), you need to tell Craft, which fields you are going to use. In this case, we will need the `imageField`. Please note, that this can be a comma separated list (Array syntax) and you can have more than one field here.
+* We also need to change the image selector from `.one()` to `[0]`
+
+With Eager Loading, no matter how many entries you have to fetch, Craft only needs 3 database queries instead of N+1 (`N` = number of entries + `1` = initial entries query).
+
+* Make sure you also read through the [Craft 3 Documentation](https://docs.craftcms.com/v3/dev/eager-loading-elements.html#eager-loading-image-transform-indexes)
+* If you want to dive deeper, here is a really good & detailed read about [Eager loading](https://nystudio107.com/blog/speed-up-your-craft-cms-templates-with-eager-loading).
+
+### Conclusion
+Well of course this heavily depends on the page you are serving. But in general it is a good idea to have  the first contentful paint (FCP) below 2s. If you are higher than that, does not mean you are serving a bad website, but this means that your user will lose interest very quickly. As you can see in the illustration, the bounce rate increases slowly after 2 seconds and rapidly above 3 seconds.
+
+[![Does Page Load Time Really Affect Bounce Rate?](http://res.cloudinary.com/dsteinel/image/upload/c_scale,w_512/v1532351329/Espi/blog/performance%20profiling/relation__page-speed--bounce-rate.png)](https://royal.pingdom.com/2018/01/18/page-load-time-really-affect-bounce-rate/)
+
+If you encounter that your page is having some performance issues, Eager Loading elements will improve the speed of your website. 
+After you optimised your template, the next step would be to enable caching. Craft has its build-in [caching tag](https://docs.craftcms.com/v3/dev/tags/cache.html) with which you can cache specific parts of your template.
+The [Blitz Plugin](https://github.com/putyourlightson/craft-blitz) is a intelligent static file caching which will improve the first contentful paint even more.
+
+Do not only check your page-speed with the Craft debugger. Also check the network tab in your [browser inspector](https://developers.google.com/web/tools/chrome-devtools/network-performance/) and also use tools like these one:
```

Comment:
I would love to read an example with concrete figures how this has improved page speed in one of our projects (Friday Magazine for instance)